homeassistant/components/modbus/sensor.py

```diff
@@ -141,10 +145,12 @@ async def async_setup_platform(
             _LOGGER.error("Error in sensor %s structure: %s", entry[CONF_NAME], err)
             continue
 
-        if entry[CONF_COUNT] * 2 != size:
+        if regSize != size:
             _LOGGER.error(
-                "Structure size (%d bytes) mismatch registers count (%d words)",
+                "Structure size (%d bytes) mismatch registers size (%d bytes). Multi: %d, Count: %d",
                 size,
+                regSize,
+                entry.get(CONF_REGISTER_SIZE, 1),
```

Comment:
I am lost between regSize and entry[CONF_REGISTER_SIZE] as the names say they should be the same.